Automattic/wp-calypso
# Onboarding Checklist: Pull out all config flags & make changes needed to work in the Desktop env

The top-level config flag wasn't enabled in horizon.  Remedying that in #34446, but this goes further to pull the flags out of all environments since it's enabled nearly everywhere anyway.

Interestingly, the `onboarding-checklist` flag is [currently disabled](https://github.com/Automattic/wp-calypso/blob/ad190268ff0c526773ee0b7c5d3a6805b861d6b3/config/test.json#L75) in the test environment.  I'm not sure if this is intentional and what issues it will cause to enable it there.

Enabling `onboarding-checklist/phase2` in the desktop app exposed an issue where the checklist page failed to load in the app. This PR also addresses this issue.

#### Changes proposed in this Pull Request

* rm the flags in config files
* rm the `config.isEnabled` checks
* work around checklist loading issue by ensuring response is unpacked from http envelope

#### Testing instructions

* Creating a new site on this branch lands the user on the checklist page, and the checklist works
* Once merged to master, creating a new site on `horizon` lands the user on the checklist page, and the checklist works
* Building wp-desktop with this branch, creating a new site lands the user on the checklist page, and the checklist works (see below)

#### Test this branch in wp-desktop

* Stop any running instances of calypso (to free the network port)
* Clone https://github.com/Automattic/wp-desktop/ & check out `develop`
* [Set up your secrets](https://github.com/Automattic/wp-desktop/blob/develop/docs/secrets.md)
* cd into the `calypso` subdirectory in the wp-desktop dir
* `git checkout remove/onboarding-checklist-flags` (this branch)
* cd back to the top level wp-desktop dir
  * `make dev-server`
* Once the above is ready, in another terminal, cd to the top level wp-desktop dir
  * `make dev`
* You should see calypso inside an electron wrapper

##### Test an existing site in the desktop app

* Open the dev tools (View|Developer Tools)
* Enter this in the console: `document.location = '/checklist/yourwpcomtestsiteslug';`
* You should see the checklist, items should be populated, items should be completable in the same manner on desktop as on the production web

##### Test a new site in the desktop app

* Switch site | Add new site
* Create a shiny new...
* Complete the flow
* You should see the checklist, items should be populated, items should be completable in the same manner on desktop as on the production web

Fix for Automattic/wp-desktop#641 -- once the submodule is bumped
